pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -0,0 +1,380 @@
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource ComplianceRemediation
+	err = c.Watch(&source.Kind{Type: &complianceoperatorv1alpha1.ComplianceRemediation{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileComplianceRemediation implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileComplianceRemediation{}
+
+// ReconcileComplianceRemediation reconciles a ComplianceRemediation object
+type ReconcileComplianceRemediation struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client   client.Client
+	mcClient *mcfgClient.MachineconfigurationV1Client
+	scheme   *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a ComplianceRemediation object and makes changes based on the state read
+// and what is in the ComplianceRemediation.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileComplianceRemediation) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	var err error
+
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling ComplianceRemediation")
+
+	// Fetch the ComplianceRemediation instance
+	remediationInstance := &complianceoperatorv1alpha1.ComplianceRemediation{}
+	err = r.client.Get(context.TODO(), request.NamespacedName, remediationInstance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	if remediationInstance.Spec.Apply == false {
+		reqLogger.Info("Won't reconcile remediation which is not selected", "remediation", request.NamespacedName)
+		return reconcile.Result{}, nil
+	}
+
+	if remediationInstance.Status.ApplicationState == complianceoperatorv1alpha1.RemediationApplied {
+		reqLogger.Info("Won't reconcile remediation which is already applied", "remediation", request.NamespacedName)
+		return reconcile.Result{}, nil
+	}
+
+	switch remediationInstance.Spec.Type {
+	case complianceoperatorv1alpha1.McRemediation:
+		reqLogger.Info("Reconciling a MachineConfig remediation")
+		err = r.reconcileMcRemediation(remediationInstance, reqLogger)
+	default:
+		err = fmt.Errorf("Unsupported remediation type %s", remediationInstance.Spec.Type)
+	}
+
+	// this would have been much nicer with go 1.13 using errors.Is()
```

Comment:
Trying to address it https://github.com/openshift/compliance-operator/pull/50

Reply:
Nope, no 1.13 builder image yet.

Reply:
yup, that's why I didn't end up using errors.Is :)